src/solver/mlir_common.cpp

```diff
@@ -41,56 +45,74 @@ std::string InsertGToLayout(const std::string& layout, char dim)
     return layout_with_g.insert(index, 1, 'G');
 }
 
+const char* DTypeName(miopenDataType_t ty)
+{
+    switch(ty)
+    {
+    case miopenHalf: return "fp16";
+    case miopenFloat: return "fp32";
+    case miopenDouble: return "fp64";
+    case miopenBFloat16: return "bf16";
+    case miopenInt32: return "i32";
+    case miopenInt8: return "i8";
+    case miopenInt8x4: return "i8x4";
+    }
+    MIOPEN_THROW("Internal error: value outside of datatype enum");
```

Comment:
[Recommendation] Here's what I meant:
```suggestion
    MIOPEN_THROW(miopenStatusInternalError, "Value outside of datatype enum");
```